src/charge_controller/dev/LCDDisplay.cpp

```diff
@@ -1,6 +1,70 @@
 
 #include <charge_controller/dev/LCDDisplay.hpp>
+#include <cstdio>
 
-LCDDisplay::LCDDisplay(IO::SPI &spi, IO::GPIO &reset, IO::GPIO &A0) {}
+LCDDisplay::LCDDisplay(IO::GPIO& reg_select, IO::GPIO& reset, IO::SPI& spi) :
+lcd(DEV::LCD(reg_select, reset, spi, 12, 3, 2)) { }
 
-void LCDDisplay::init() {}
+void LCDDisplay::init() {
+    lcd.initLCD();
+    lcd.clearLCD();
+    lcd.setDefaultSections(SECTION_TITLES);
+    showSplashScreen();
+}
+
+void LCDDisplay::showSplashScreen() {
+    lcd.clearLCD();
+    lcd.setEntireScreenBitMap(evtBitMap);
+}
+
+void LCDDisplay::showSections() {
+    lcd.clearLCD();
+    lcd.displaySectionHeaders();
+}
+
+void LCDDisplay::setChargeControllerStatus(const char *str) {
+    chargeControllerStatus = str;
+}
+
+void LCDDisplay::display() {
+    lcd.setTextForSection(0, batteryOneStatus);
+
+    lcd.setTextForSection(1, chargeControllerStatus);
+
+    lcd.setTextForSection(2, batteryTwoStatus);
+
+    char batteryOneVolt[32];
+    std::sprintf(batteryOneVolt, "%d v", batteryOneVoltage);
```

Comment:
Volts use a capital V
This is quite possibly one of the nitpickiest comments I have ever left